src/util/shmem/fd_shmem_admin.c

```diff
@@ -566,13 +568,71 @@ fd_shmem_page_sz_to_cstr( ulong page_sz ) {
   return "unknown";
 }
 
+/* fd_shmem_check_rlimits: Checks whether process rlimits are configured
+   appropriately. Logs warnings to the user if any issues are found. */
+static void
+fd_shmem_check_rlimits( void ) {
+  /* Get current rlimits */
+
+  struct rlimit memlock, nice, rtprio;
+
+# define GET_RLIMIT( limit, var )                                      \
+  do {                                                                 \
+    if( FD_UNLIKELY( 0!=getrlimit( limit, var ) ) ) {                  \
+      FD_LOG_WARNING(( "getrlimit(" #limit ") failed: %s",             \
+                       strerror( errno ) ));                           \
+      return;                                                          \
+    }                                                                  \
+  } while(0)
+
+  GET_RLIMIT( RLIMIT_MEMLOCK, &memlock );
+  GET_RLIMIT( RLIMIT_NICE,    &nice    );
+  GET_RLIMIT( RLIMIT_RTPRIO,  &rtprio  );
+
+# undef GET_RLIMIT
+
+  FD_LOG_INFO(( "RLIMIT_MEMLOCK: soft=%lu hard=%lu", memlock.rlim_cur, memlock.rlim_max ));
+  FD_LOG_INFO(( "RLIMIT_NICE:    soft=%lu hard=%lu", nice   .rlim_cur, nice   .rlim_max ));
+  FD_LOG_INFO(( "RLIMIT_RTPRIO:  soft=%lu hard=%lu", rtprio .rlim_cur, rtprio .rlim_max ));
+
+  /* Get sysinfo */
+
+  struct sysinfo info;
+  if( FD_UNLIKELY( 0!=sysinfo( &info ) ) ) {
+    FD_LOG_WARNING(( "sysinfo() failed: %s", strerror( errno ) ));
+    return;
+  }
+  FD_LOG_INFO(( "sysinfo().totalram = %lu", info.totalram ));
```

Comment:
sysinfo is a Linux-specific API.  It'd be preferable to use a POSIX equivalent (if there is one in POSIX <= POSIX-2008) for better cross platform portability.  Not sure if there are good options for that though.